### ISSUE TITLE
Newest Docker API changes break ps behaviour

### ISSUE THREAD

**Issue by moritzheiber** [xeger/docker-compose#58]
The newest Docker API changes are breaking the size calculation in [`container.rb`](https://github.com/xeger/docker-compose/blob/master/lib/docker/compose/container.rb#L24-L29). The string literal returned for `{{.Size}}` is now `(xxB (virtual xxMB))`, instantiating a new session ends with:

```
'Service#new' failed with status 'virtual': Impossibly large unit
```

**Comment by xeger** [comment]
Grr! This is the downside of pretending that the CLI is an API. I will fix over the weekend or so; thanks for reporting.

**Comment by moritzheiber** [comment]
Given that you're actually using the `docker` CLI here instead of `docker-compose`, would it be a viable alternative to call the Docker API directly? The `docker-api` gem would have you covered and you wouldn't need to do any custom parsing etc.

It would be a lot less crap to take care of or that could possibly break from version to version.

**Comment by xeger** [comment]
That's a fair point. In the long run, I would like this gem to become a _reimplementation_ of Compose as opposed to a wrapper -- which would require extensive docker API work. Not sure I want to add a new dependency for this one case, but if it is consistent with my long term plan ... it's a bit attractive.

**Comment by xeger** [comment]
Published a fix as 1.1.5; let me know if you encounter any further difficulty.

I'll put some thought into your comment about API vs. CLI, but I'd need to figure out the future of the `docker` gem (which I also maintain, and which is basically not useful for the past several years, now that Docker has a supported REST API). I'll also need to look at `docker-api`, its dependencies and its Ruby API ... if the design is sound, there is an argument to be made that I should "donate" the docker gem name to that project.

If this gem becomes a hybrid that sometimes uses `docker-compose-api` and sometimes uses CLI .. well, I suppose there are still some advantages, especially if we use API for "read" type operations whose output can change frequently. On the other hand, there's the K.I.S.S. argument; for example, given the status quo, I don't need to concern myself with TLS client auth; the docker binary handles that for me.
